airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_read_processor.py

```diff
@@ -175,6 +176,17 @@ def start_next_partition_generator(self) -> Optional[AirbyteMessage]:
         else:
             return None
 
+    @staticmethod
+    def _generate_failed_streams_error_message(stream_failures: Mapping[str, List[Exception]]) -> str:
```

Comment:
not something i'd block on, but i think it would be nice to consolidate this in a helper utility in once placesince its basically a copy of the one in `abstract_source.py`.

Except this one supports a list of exceptions per stream. But given this fix feels more urgent I'd say we don't need to fix it in this PR. The one in `abstract_source.py` could still put it into a list and call this and it will just always be a list of one if we wanted to reuse.